5-infrastructure/modules/base-env/boa-sec.tf

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  kms_settings = {
```

Comment:
this does not give us benefits of using for_each since we are statically defining each objects
we should use something like
```tf
locals {
    kms_settings = {for c in setproduct(["gke", "sql"], ["us-central1", "us-east4"]): "${c[0]}-${c[1]}" => {product = c[0], location = c[1]}}
}


module "kms_keyrings_keys" {
  source   = "terraform-google-modules/kms/google"
  version  = "~> 2.0"
  for_each = local.kms_settings

  project_id           = "foo"
  prevent_destroy      = false
  key_protection_level = "HSM"


  location           = each.value.location
  keyring            = "kms-ring-boa-${each.value.location}-${each.value.product}-01"
  keys               = ["kms-key-h-boa-${each.value.location}-${each.value.product}-01"]
....
}
```

Reply:
yup was testing this out already for later, will push update momentarily

Reply:
@bharathkkb please check 506a8778168aee87e3473f9122ed44cee4a6a2c0